src/Processors/QueryPlan/ReadFromMergeTree.cpp

```diff
@@ -1646,8 +1647,15 @@ static void buildIndexes(
             condition = index_helper->createIndexCondition(filter_actions_dag, context);
         }
 
-        if (!condition->alwaysUnknownOrTrue())
+        if (const auto * bloom_filter_index_condition = typeid_cast<const MergeTreeIndexConditionBloomFilter *>(condition.get()))
```

Comment:
The special case for vector similarity indexes above is already not pretty and I am the person to blame for it.

But we should avoid new special casing. I wonder why the new logic for bloom filter indexes cannot be moved into `alwaysUnknownOrTrue` so that we don't need to change the existing code?